rfc/rfc017-distributed-network-grpc-v2.md

```diff
@@ -54,26 +54,116 @@ As specified by RFC015, the node MUST authenticate the peer's node DID as follow
+
+3. Alice responds with the requested transactions (§5.2.3).
+
+4. After adding Alice's transactions to the DAG (making sure its cryptographic signature is valid), Bob SHOULD query the payload if it's missing.
+
+A node MUST make sure to only add transactions of which all previous transactions are present. If previous transactions are missing, Bob will send a `State` message.
+
+#### 5.2.1 Broadcasting new transactions
+
+A node's new transaction references MUST be broadcast at an interval using the `Gossip` message, by default every 2 seconds. The interval MAY be adjusted by the node operator but MUST conform to the limits (min/max interval) defined by the network. It is advised to keep it relatively short because it directly influences the speed by which new transactions are retrieved.
+
+The `Gossip` message MUST contain an `XOR` value, an `LC` value and a list of transaction references (`transactions`).
+The list MUST contain all new transaction references since the last `Gossip` message.
+The list of transaction references MUST be tracked per connection.
+If no new transactions have been added/received or for the first message for a connection, an empty list is sent.
+The list MUST not contain more than 100 transactions.
+This could create a backlog of messages at the sending node's side. 
+A node SHOULD take precautions on keeping this backlog to a minimum, eg it SHOULD reduce the inflow of new transactions.
+
+The `LC` value MUST equal the highest Lamport Clock value of all transaction references included in the `XOR` calculation.
+If no transactions are present, an all-zero `XOR` and `LC` of 0 is sent.
+
+This message does not require a `conversationID`
+
+#### 5.2.2 Transaction List Query
+
+Upon receiving a `Gossip` message from a peer, a node MUST first compare the `XOR` value from the message with its own `XOR` value from the DAG.
+If the values are equal, no further action is required.
+
+If they are not equal, the transaction list MUST be filtered. All known transaction references MUST be removed.
+The resulting list contains all transaction the node is missing.
+The node MUST send a `TransactionListQuery` message containing the list of missing transaction references.
+This message MUST also add a `conversationID`.
+
+If the resulting list of transactions is empty and the `LC` value in the message is equal to or higher than the highest Lamport Clock value of the node, then the node MUST send a `State` message. See §6.2.1.
+
+#### 5.2.3 Transaction List
+
+When a node receives a `TransactionListQuery` message, it MUST respond with a `TransactionList` message.
+This is a response type message so it MUST include the sent `conversationID`.
+Each transaction in the message MUST have been requested. Transactions that resulted from a `TransactionListQuery` MUST have been present in that message.
```

Comment:
stating the same thing twice, in other wording? Better write it once, but then in such a way it doesn't require a second wording.